scheduler/filter/utils.go

```diff
@@ -0,0 +1,89 @@
+package filter
+
+import (
+	"regexp"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+type comparison int
+
+const (
+	equ = comparison(iota)
+	neg
+	gte
+	lte
+)
+
+func parse(k, v string) (string, string, comparison, bool) {
+	// default comparison mode
+	mode := equ
+
+	// support case of constraint:k==v
+	// try to check = on both sides, "k= : = : v" and "k : = : =v", to make sure it works everywhere
+	if strings.HasSuffix(k, "=") {
+		k = strings.TrimSuffix(k, "=")
+	} else if strings.HasPrefix(v, "=") {
+		v = strings.TrimPrefix(v, "=")
+	}
+
+	if strings.HasPrefix(v, "!") {
```

Comment:
I see. Thank you for your comments. I'll refactor this next time I add another operator.
Which operator do you think we should add more anyway?